client/server/config/parser.js

```diff
@@ -66,7 +66,21 @@ module.exports = function ( configPath, defaultOpts ) {
 		data.features[ 'wpcom-user-bootstrap' ] = false;
 	}
 
-	const serverData = assign( {}, data, getDataFromFile( secretsPath ) );
+	const secrets = getDataFromFile( secretsPath );
+	if ( opts.env === 'desktop' && secretsPath === realSecretsPath ) {
+		const oauthClientId = secrets.desktop_oauth_client_id;
+		const oauthSecret = secrets.desktop_oauth_client_secret;
+		const oauthTokenEndpoint = secrets.desktop_oauth_token_endpoint;
+
+		data.oauth_client_id = oauthClientId !== undefined ? oauthClientId : false;
```

Comment:
This can be cleaned up a bit:

data.oauth_client_id = secrets.desktop_oauth_client_id ?? false